### ISSUE TITLE
Add vignette to compare with {finalsize}

### ISSUE THREAD

**Issue by pratikunterwegs** [epiverse-trace/epidemics#33]
This PR is to request a vignette that compares the final size calculations from {epidemics} against those from {finalsize}, to show the effect of implementing interventions and vaccination regimes.

**Comment by BlackEdder** [comment]
WIP: https://github.com/epiverse-trace/epidemics/tree/finalsize_vignette

**Comment by BlackEdder** [comment]
The vignette includes a couple of simple code examples to compare the two approaches. I noticed a couple of differences in the implementation that might be good to think about.

- There are some naming differences, particularly demo_grp vs demography_group
- The form of the contact matrix used is different in both packages. For SEIR-V it is not needed to rescale the contact matrix, presumably because it does so internally. In finalsize this is needed. Would be good if we can think of a way to unify this.
- Is there a general way to know the proportion of the population vaccinated in the vaccine calendar? It did not seem to be equal to nu times the number of days the vaccination programme is happening?

**Comment by adamkucharski** [comment]
Have had a look through, and useful to see them side-by-side to understand structure/naming etc. A few additional thoughts:
- If including as vignette, would be good to distinguish what it adds beyond the existing [compartmental comparison](https://epiverse-trace.github.io/finalsize/articles/compare_sir_model.html) in finalsize.
- Perhaps an obvious one is alignment in how different aspects are defined. Perhaps too much refactoring to get full alignment, would be nice if possible to unify parameters so easier to use in the other model.
- Another would be the difference in speed, particularly if sampling over many realisations of R0 (as discussed in [finalsize](https://epiverse-trace.github.io/finalsize/articles/uncertainty_params.html) and issue #106)
- There's also quite a lot of wrangling code required to compare, so could be useful to have this vignette as more of a guide to help users decide which is preferable, e.g. if you want to easily calculate overall size, and do multiple realisations, finalsize gets you there much faster and potentially more accurately, because it's based on analytical solution. Whereas any temporal questions have to be addressed with epidemics?

**Comment by pratikunterwegs** [comment]
> The vignette includes a couple of simple code examples to compare the two approaches. I noticed a couple of differences in the implementation that might be good to think about.

Thanks @BlackEdder, will you keep working on this vignette or should I take this further and make a PR?

>     * There are some naming differences, particularly demo_grp vs demography_group

Thanks - maybe 'demo_group' is a good compromise between the two packages, and between clarity and brevity. That said {finalsize} is on CRAN so perhaps adopting 'demo_grp' in {epidemics} is safer than changing it in both packages.

>     * The form of the contact matrix used is different in both packages. For SEIR-V it is not needed to rescale the contact matrix, presumably because it does so internally. In finalsize this is needed. Would be good if we can think of a way to unify this.

Yes, it is scaled internally using `.prepare_args_default()`. This could be a good time to take another look at the {contactmatrix} package we had been discussing?

>     * Is there a general way to know the proportion of the population vaccinated in the vaccine calendar? It did not seem to be equal to nu times the number of days the vaccination programme is happening?

I had not really thought about it, but I can convert it to an issue and a relevant test for correctness.

**Comment by pratikunterwegs** [comment]
> Have had a look through, and useful to see them side-by-side to understand structure/naming etc. A few additional thoughts:
> 
>     * If including as vignette, would be good to distinguish what it adds beyond the existing [compartmental comparison](https://epiverse-trace.github.io/finalsize/articles/compare_sir_model.html) in finalsize.
> 
>     * Perhaps an obvious one is alignment in how different aspects are defined. Perhaps too much refactoring to get full alignment, would be nice if possible to unify parameters so easier to use in the other model.
> 
>     * Another would be the difference in speed, particularly if sampling over many realisations of R0 (as discussed in [finalsize](https://epiverse-trace.github.io/finalsize/articles/uncertainty_params.html) and issue [<infection> accepts R as a distribution or samples #106](https://github.com/epiverse-trace/epidemics/issues/106))
> 
>     * There's also quite a lot of wrangling code required to compare, so could be useful to have this vignette as more of a guide to help users decide which is preferable, e.g. if you want to easily calculate overall size, and do multiple realisations, finalsize gets you there much faster and potentially more accurately, because it's based on analytical solution. Whereas any temporal questions have to be addressed with epidemics?

Thanks @adamkucharski - I think being able to answer the temporal questions, and the effect of interventions etc is probably the key aspect. I can add those if not already present.

Speaking of temporal questions, would it be useful to have helper functions that help get the timing and size of the epidemic peak (optionally by age group)?

**Comment by adamkucharski** [comment]
Yep, size in particular often useful as output metric, particularly for healthcare requirements, deaths etc. (e.g. Table 2 in [early COVID modelling](https://www.thelancet.com/journals/lanpub/article/PIIS2468-2667(20)30133-X/fulltext)) and tables in [Roadmap analysis](https://www.gov.uk/government/publications/lshtm-updated-roadmap-assessment-prior-to-delayed-step-4-7-july-2021).

**Comment by pratikunterwegs** [comment]
Thanks - we do have `epidemic_size()` that calculates size at any point in the simulation, but I'll put in some functions for the peak specifically.

**Comment by BlackEdder** [comment]
> > The vignette includes a couple of simple code examples to compare the two approaches. I noticed a couple of differences in the implementation that might be good to think about.
> 
> Thanks @BlackEdder, will you keep working on this vignette or should I take this further and make a PR?

Might be most efficient if you would be willing to take this further. Agreed that it would be good to explain in the text, what the main differences are between the two packages. I would hazard that final size is faster, but if you need dynamics over time and/or interventions/vaccinations that happen over time, you should use the epidemics package.

**Comment by pratikunterwegs** [comment]
Okay thanks, will do.
